### PR TITLE
Nahiayn_New Dark Mode Button

### PR DESCRIPTION
**Pull request by Nahiyan-16**
# Description
Created a brand new dark mode button with hover effects

## Related PRS (if any):
This frontend PR is related to the dev backend PR.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to dashboard→ "Welcome, User Name" -> Check if the dark mode toggle is working properly

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/f32790f8-7f9e-4ec1-8fda-7a0cd0d96f0d

### CODE REVIEW

**Review by kdaga81 — DISMISSED** [review]
image
image

This looks awesome! Good work!

**Review by SrikanthChowdary495 — DISMISSED** [review]
I have tested this PR. It is working in both Darkmode & lightmode is working.
dark mode
ligt mode

**Review by kabraambika — CHANGES_REQUESTED** [review]
I reviewed the feature on PR is working as intended but I noticed that if I click before the animation finishes, the image appears to get stuck on the text. Please see the attached video for reference.

https://github.com/user-attachments/assets/f8b3ec2d-aea9-4b1f-a3a2-7d54b9923ffe

**Review by Parth-tech — DISMISSED** [review]
Hey Nahiayan,
The PR works as intended but as @kabraambika highlighted try dealing with the warning.
Approving the PR as the required functionality is met.
Screenshot 2024-07-19 at 8 28 03 PM

**Review by Chetan0606 — APPROVED** [review]
Verified as per the provided steps, It looks good.
Screenshot (314)

**Review by angelalalacheng — DISMISSED** [review]
Hi, 
I use the latest development backend branch and your frontend branch to test. When the mouse hover on button, the animation works good. And I also test the mobile device and it all works without animation.

https://www.loom.com/share/3a1defb61fa644f7bb876fd9213ef5c4?sid=a94e7699-729c-40a8-b143-1115813a427f

Screenshot 2024-07-20 at 9 49 26 AM

**Comment by one-community** [comment]
Thank you all, merging!

**Comment by souravwalke** [comment]
I tested the feature and found a small UI issue when toggling from Dark mode to Light mode. The text that should display as "Light Mode" does not appear as it gets masked by the background color. I've attached a screenshot for you to look over.

image

https://www.loom.com/share/9136e2bda48d4c42851bf19b9fccce38?sid=e34f7dce-5b45-48c4-9f13-181a88f9c024

**Review by pavanswaroopl — COMMENTED** [review]
Great work,

Have some observations, light mode text doesn't appear at times, simulation is upload below.

https://github.com/user-attachments/assets/44316764-1526-4e1c-8a09-92c9d7d59663

Screenshot 2024-07-20 at 9 19 58 PM

File: src/components/Header/DarkMode.css [hunk 1]
```diff
@@ -16,4 +16,140 @@
...
+}
+
+/* DARK MODE BUTTON */
+
+.darkModeSliderContainer {
+  width: 120px;
+  height: 30px;
+  border-radius: 20px;
+  display: flex;
+  justify-content: start;
```
[the remark this answers is not in the document]
Comment (kabraambika) [reply src/components/Header/DarkMode.css]: Thanks for resolving in new changeset 05143d20db8ee0676472113d77a4e8f8dcce6788.

**Review by kabraambika — COMMENTED** [review]
The compilation warnings have been addressed. However, there are still two issues to resolve:

- The "Light mode" text is not visible when dark mode is enabled as mentioned by @souravwalke and @pavanswaroopl  
- If the button is clicked before the animation completes, the icon becomes stuck on the text

**Pull request merged**